### ISSUE TITLE
authentication (login, register, reset password)

### ISSUE THREAD

**Issue by rkoberg** [Shopify/js-buy-sdk#214]
Do you really need Shopify Plus to be able to log a user in from a shop owner's pre-existing web site? From what I read in the forums, shopify plus costs $1200/month! 

I can create a customer through the rest API by proxying the request through my web site's server side code. I just want to have them login, and preferably checkout through my (client's) web site. Alternatively, is there a way to run a an isomorphic node/react webapp under the shopify authentication domain?

**Comment by berkaey** [comment]
+1

**Comment by zstrad44** [comment]
+1 bump

**Comment by minasmart** [comment]
Hey! Parts of these things are roadmapped. We'll have the ability to let a user log in and associate a cart with their account some time in the new year. As for checking out on any 3rd party site, we may enable that at some point. For now, there are a lot of PCI concerns around allowing a 3rd party to collect and forward credit card data. There is some work planned, but nothing that I can talk about as of yet.

Sorry guys!

**Comment by christianseel** [comment]
Hi @minasmart, is there any update on that? Is that sill on the roadmap? Any more precise ETA?
I also have an existing login on my existing website and would like to pass that user object to shopify and/or store the user id in the checkout/order.

**Comment by minasmart** [comment]
Yep! We've been doing some significant refactoring of our APIs. When the new features are available, they'll be pushed up to this repo likely in a beta branch to start. All issues asking for these features will be updated when that happens.

**Comment by skywalkthis** [comment]
@minasmart any update on this? Specifically the ability to perform some variation of Multipass for non Shopify Plus accounts? Thanks!

**Comment by jpdesigndev** [comment]
@mikkoh @minasmart

I’ve been searching all over for some docs on authenticating users in lieu of Shopify Plus + Multipass. I’m only commenting here because I’m hoping for something more concrete on the feasibility of this landing this year. I don’t see anything on the beta or alpha branches, so I’m hoping someone at Shopify can comment.

**Comment by minasmart** [comment]
This functionality is available in our storefront API. There's [a mutation](https://help.shopify.com/api/storefront-api/reference/mutation/customeraccesstokencreate) that allows you to log in a customer, then there are several fields and other mutations you can access within that API using the token you've created.

Unfortunately, we don't have a cohesive guide as of yet.

**Comment by ericpoon** [comment]
I really need this feature so currently I'm trying to add customer auth/management features by myself. I have made some progress (can successfully generate/delete/renew customer access token, create new account, and create shipping address). Please see my [Github](https://github.com/ericpoon/js-buy-sdk) if you are interested.

**Comment by timbroder** [comment]
@minasmart did any kind of guide for this surface? Thanks!

**Comment by smashah** [comment]
What's the status on this? Maybe at least write in the docs that auth is to be implemented by ourselves? Is there any advice you can share on this? I need to 1. authenticate user and 2. attach customer to a cart.

**Comment by lucasprag** [comment]
Any news @minasmart ? Thanks

**Comment by abishekrsrikaanth** [comment]
@rebeccajfriedman any updates on this?

**Comment by ThomasT404** [comment]
+1

**Comment by arobbins** [comment]
+1

**Comment by damienroche** [comment]
4 years after, and we still can't associate a checkout to a customer. It's a must have enhancement to switch some stores to shopify. any chances to have an update about this @minasmart ?

**Comment by arobbins** [comment]
I have thousands of users who would love this feature enabled, specifically the ability to attach a customer to a cart session. I'm guessing there are security implications to this? Could we do it via the sever instead of the client?

**Comment by josephchan91** [comment]
+1
